### DISCUSSION TITLE
TiNetz meter cannot be read

### DISCUSSION THREAD

**Discussion by burenziege** [UtilitechAS/amsreader-firmware#618] [General]
Hallo
 I would first like to commend you for the excellent work.

Now to my problem. I have a working installation running on the Rpb 4. But I want to use AMSreader. 
Unfortunately, AMSreader cannot decrypt.

Hardware:
D1 mini, M-Bus to UART converter MIKROE TSS721AR

AMS reader error message:
HAN: Unknown data received, check meter config

TiNetz meter Kaifa MA309M settings:

Serial conf.  inverted > NO
2400  8N1 256
Voltage  400V (TN)
Main fuse 40A Production 10kWp
Meter is encrypted Yes (Key)
Authentication key No (0000000000000000)
Multipliers No

How can I set MBUS like with Netz Niederösterreich?
(Kaifa MA309M | 2400 | 8N1 | Netz Niederösterreich GmbH | Yes (Encryption key only) | MBUS | #433)

Are the settings for the TiNetz meter the same as for Netz Niederösterreich?
I think there is a difference but I can't find it.

Vielen Dank für die ausgezeichnete Arbeit
Christoph

**Comment by Noschvie** [comment]
For TiNetz you need an optical interface, IR interface. Have a look to the documents. It‘s not M-bus like Netz Niederösterreich… 
Like this https://wiki.volkszaehler.org/hardware/controllers/ir-schreib-lesekopf-ttl-ausgang

**Comment by Noschvie** [comment]
Maybe we should start a project to support all the different SmartMeter of the austrian energy supplier…

**Comment by ArnieO** [comment]
@burenziege : 
Thank you for your kind words!

1. Control question: Are you sure your M-bus conversion circuitry works correctly?
2. Please share screenshot of your config page
3. Please run a Telnet Debug: https://amsleser.no/blog/post/24-telnet-debug
and show us a screendump of the output

If you have access to technical documentation for your meter, it could help us understanding how to support it. (German documentation is fine!)

@Noschvie : As he mentions an M-bus conversion, I do not think this is a meter with (only) optical eye.

**Reply by Noschvie** [reply]
Find below the documentation, which means M-bus
https://www.tinetz.at/uploads/tx_bh/tinetz_smart-meter_beschreibung-kundenschnittstelle_001.pdf

Optische Schnittstelle: wird ausschließlich zu Wartungszwecken verwendet

**Reply by ArnieO** [reply]
Thank you!
So this is clearly an M-bus meter, and the optical port is for maintenance only.
Parity is (if my German is not too rusty) 8E1, which is what we are used to on M-bus meters:
![image](https://github.com/UtilitechAS/amsreader-firmware/assets/10295178/bf1b1750-2e30-47e4-9932-5070b99e0f5e)

A [Pow-U](https://amsleser.no/hjem/7-pow-u-universal.html) should work just fine on this meter. 

Then we wait for @burenziege to check if his M-bus conversion works, and to send us a Telnet Debug printout.

**Comment by burenziege** [comment]
Thank you for the quick help!  
I won't be home until next week.  then I will provide the requested data.  

I believe that my m-bus is working properly, because I already provide the counter with the Raspberry using the same hardware.  but I'll know more next week.  

kind regards christoph

**Reply by ArnieO** [reply]
Right; you got it working on a RasPi.
It could maybe be useful if you then also provide a wiring schematic - just to get an extra eye on it. 
If you search for Wemos D1 mini on the "Issues" here you will se many that have had difficulties getting it running correctly on that hardware. There are many clones on the market, and my usual suspect is inadequate decoupling or other issues related to power supply of Wemos and/or M-bus conversion module.

I'll try to help as far as I can providing you give a bit more information on your setup.

**Comment by burenziege** [comment]
filter:
(W) Used 1990ms to read HAN port (true)
(W) loop() used 2012ms
(W) No HAN data received last 30s, single blink
(D) Serving /data.json over http...
(V) MBUS frame:
(V) 68 68 01 67
(W) Boundry flag missing
(W) Used 1961ms to read HAN port (true)
(D) Serving /data.json over http...
(D) Serving /data.json over http...
(W) No HAN data received last 30s, single blink
(V) MBUS frame:
(V) 68 68 01 67
(W) Boundry flag missing
(W) Used 1971ms to read HAN port (true)
(D) Serving /data.json over http...
(D) Serving /data.json over http...
(V) MBUS frame:
(V) 68 68 01 67
(W) Boundry flag missing
(W) Used 1994ms to read HAN port (true)
(W) loop() used 2016ms
(W) No HAN data received last 30s, single blink
(D) Serving /data.json over http...
(D) Serving /data.json over http...
(V) MBUS frame:
(V) 68 68 01 67
(W) Boundry flag missing
(W) Used 1958ms to read HAN port (true)

**Reply by gskjold** [reply]
This looks to me like it is loosing data in between so that it is not getting full frames. A wiring diagram would be useful. Maybe you have used UART0 on your D1? If so, this will be the reason, as a USB-TTL chip is connected to that UART which will interfere with your data.

**Comment by burenziege** [comment]
Hello to the community!
Now I have some time and got it working.

My settings for TiNetz:

amsconfig
version v2.2.21
board type 101 (Wemos d1 mini)
ip xxx.xxx.....
gateway xxx.xxx.....
subnet 255.255.255.0
dns1 xxx.xxx.....
mdns 1
mqttHost xxx.xxx.....
mqttPort 1883
mqttClientId tinetzaehler
mqttPublishTopic tinetzaehler
mqttPayloadFormat 4
mqttSsl 0
homeAssistantDiscoveryPrefix homeassistant
homeAssistantDiscoveryHostname tinetzaehler
homeAssistantDiscoveryNameTag tinetzaehler
meter baud 2400
meter Parity 8E1
meterInvert 0
meter distribution system 2
meter MainFuse 25
meter Production Capacity 10
gpioHanPin 13
gpioHanPinPullup 0
gpioApp Pin 4
gpioLedPin 2
gpioLedInverted 1
gpioVccOffset 0.00
gpioVccMultiplier 1,100
gpioVccBootLimit 0.0
ntpEnable 1
ntpDhcp 1
ntpTimezone Europe/Vienna
ntpServer xxx.xxx.....
entsoeArea 10YAT-APG------L
entsoeCurrency EUR
entsoeMultiplier 1,000
entsoeFixedPrice 0.000
thresholds 5 10 15 20 25 50 75 100 150 65535 3
